pkgs/tools/misc/youtube-dl/default.nix

```diff
@@ -15,13 +15,10 @@
 buildPythonPackage rec {
 
   pname = "youtube-dl";
-  # The websites youtube-dl deals with are a very moving target. That means that
-  # downloads break constantly. Because of that, updates should always be backported
-  # to the latest stable release.
   version = "2021.06.06";
 
-  src = fetchurl {
-    url = "https://yt-dl.org/downloads/${version}/${pname}-${version}.tar.gz";
+  src = fetchPypi {
+    inherit pname version;
     sha256 = "1hqan9h55x9gfdakw554vic68w9gpvhblchwxlw265zxp56hxjrw";
```

Comment:
Did you verify this hash?

Reply:
Yeah the hash is certainly wrong because I have checked the two archives are different. But Nix is not rebuilding things for some reason. I will mark this as draft and try to figure it out.

Reply:
You need to invalidate the hash, otherwise nix does nothing.

Reply:
Just realized I was under the impression of changing URL or something and Nix will detect the changes. I remember you [have told me](https://github.com/NixOS/nixpkgs/pull/129951#issuecomment-877893877) about this but I think I got it this time. Thanks!